### PR TITLE
[BugFix] Fix incorrect hidden_states_tensor in pd disaggregation + eagle

### PR DESCRIPTION
**Pull request by ZeldaHuang**
## Motivation
When use pd disaggregation with eagle , we should transfer the hidden states in `spec_info`, But now we transfer the  hidden states in target model's `logits_output`.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/disaggregation/prefill.py [hunk 1]
```diff
@@ -410,9 +410,14 @@ def process_batch_result_disagg_prefill(
                     last_hidden_index = (
                         hidden_state_offset + extend_input_len_per_req[i] - 1
                     )
-                    req.hidden_states_tensor = (
-                        logits_output.hidden_states[last_hidden_index].cpu().clone()
-                    )
+                    if self.spec_algorithm.is_eagle():
```
Comment (ShangmingCai) [inline python/sglang/srt/disaggregation/prefill.py]: I think MTP will make this condition True as well, at least it is True for Deepseek V3/R1. Is there a way to make it work for both worlds? When using MTP, does `batch.spec_info.hidden_states[i]` equal None in this case?

Comment (ZeldaHuang) [reply python/sglang/srt/disaggregation/prefill.py]: I think in MTP, we also need to transfer the `batch.spec_info.hidden_states[i]`, cause we get the get 1 token from draft prefill in P node, ` batch.spec_info.hidden_states` will be updated to draft model's hidden states in [https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/speculative/eagle_worker.py#L940](https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/speculative/eagle_worker.py#L940)

Comment (ShangmingCai) [reply python/sglang/srt/disaggregation/prefill.py]: If that's so, do we still need this `else` logic? Just wondering if the current changes fit both MTP and vanilla EAGLE's requirements? I will trigger the CI, but it would be better if you could verify whether it works for both cases.

BTW, these lines look like they might be the same thing:
https://github.com/sgl-project/sglang/blob/main/python/sglang/srt/speculative/eagle_worker.py#L1043-L1048

**Comment by ShangmingCai** [comment]
@JustinTong0323 Can you help reviewing this PR？

**Review by ShangmingCai — COMMENTED** [review]
@JustinTong0323 BTW, I am OK with this change if the goal is only to avoid raising an error. More metadata that needs to be transmitted in the PD module will be addressed in other PRs.

File: python/sglang/srt/disaggregation/prefill.py [hunk 2]
```diff
@@ -412,9 +412,14 @@ def process_batch_result_disagg_prefill(
                     last_hidden_index = (
                         hidden_state_offset + extend_input_len_per_req[i] - 1
                     )
-                    req.hidden_states_tensor = (
-                        logits_output.hidden_states[last_hidden_index].cpu().clone()
-                    )
+                    if self.spec_algorithm.is_eagle3():
```
Comment (JustinTong0323) [inline python/sglang/srt/disaggregation/prefill.py:424]: why not `spec_algorithm.is_eagle()`?
ref:
https://github.com/sgl-project/sglang/blob/99757cc3e68418f48713e69e5543a5783abcd84d/python/sglang/srt/speculative/spec_info.py#L13-L17

Comment (ZeldaHuang) [reply python/sglang/srt/disaggregation/prefill.py:424]: As mentioned in https://github.com/sgl-project/sglang/issues/10059, the error only happened in eagle3, which has three aux hidden_states in logits_output, this pr avoid raising it. As for the accept rate of first draft decode, will be fixed in other PRs.

**Review by JustinTong0323 — APPROVED** [review]
